src/opm/io/eclipse/ESmry.cpp

```diff
@@ -309,6 +363,34 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
     }
 }
 
+std::vector<std::string> ESmry::checkForMultipleResultFiles(const std::string& rootN, bool formatted) const {
+    
+    std::vector<std::string> fileList;
+    int p = rootN.find_last_of("/");
+
+    std::string pathRootN =rootN.substr(0, p); 
+    std::string fileFilter; 
+    
+    if (formatted){
+        fileFilter =  rootN.substr(p+1) + "A";
+    } else {
+        fileFilter =  rootN.substr(p+1) + "S";
+    }
+    
+    for (boost::filesystem::directory_iterator itr(pathRootN); itr!=boost::filesystem::directory_iterator(); ++itr)
+    {
+        std::string file = itr->path().filename().c_str();
```

Comment:
No need to do `c_str()` on it.

Reply:
changed to  

```C++
        std::string file = itr->path().filename().string();
```
I'm getting compiler error without  _.string()_

Reply:
> I'm getting compiler error without _.string()_

Right, the `filename()` function returns a `boost::filesystem::path` which is not implicitly convertible to a `std::string`.  Adding `.string()` performs the requisite conversion.